profile_bluesky/startup/instrument/devices/scaler.py

```diff
@@ -12,29 +12,102 @@
+        In addition, VALUE subscriptions are run.
+        Extra kwargs are ignored (for API compatibility with EpicsSignal kwargs
+        pass through).
+        Parameters
+        ----------
+        value : any
+            Value to set
+        timestamp : float, optional
+            The timestamp associated with the value, defaults to time.time()
+        metadata : dict, optional
+            Further associated metadata with the value (such as alarm status,
+            severity, etc.)
+        force : bool, optional
+            Check the value prior to setting it, defaults to False
+        """
+        self.log.debug(
+            'put(value=%s, timestamp=%s, force=%s, metadata=%s)',
+            value, timestamp, force, metadata
+        )
+
+        if float(value) <= 0:
+            raise ValueError('preset_value has to be > 0.')
+
+        if self.parent._monitor.s.name == 'Time':
+            value_put = 1e7*value  # convert to seconds
+        else:
+            value_put = value
+
+        old_value = self._readback
+        self.parent._monitor.preset.put(value_put)
+        self._readback = value
+
+        if metadata is None:
+            metadata = {}
+
+        if timestamp is None:
+            timestamp = metadata.get('timestamp', time.time())
+
+        metadata = metadata.copy()
+        metadata['timestamp'] = timestamp
+        self._metadata.update(**metadata)
+
+        md_for_callback = {key: metadata[key]
+                           for key in self._metadata_keys
+                           if key in metadata}
+
+        if 'timestamp' not in self._metadata_keys:
+            md_for_callback['timestamp'] = timestamp
+
+        self._run_subs(sub_type=self.SUB_VALUE, old_value=old_value,
+                       value=value, **md_for_callback)
 
 
 class LocalScalerCH(ScalerCH):
-    
-    def __init__(self,*args,**kwargs):
-        super().__init__(*args,**kwargs)
-        self._preset_value = 0
-        self._monitor = self.channels.chan01 #Time is the default monitor
-    
+
+    preset_time = None
+    preset_monitor = Component(PresetMonitorSignal, kind=Kind.config)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._monitor = self.channels.chan01  # Time is the default monitor
+
     @property
     def hints(self):
         fields = []
-        for component in scalerd.channels.component_names: 
-            channel = getattr(scalerd.channels,component) 
-            if channel.kind.value in [5,7]:
+        for component in scalerd.channels.component_names:
+            channel = getattr(scalerd.channels, component)
+            if channel.kind.value in [5, 7]:
                 if len(channel.s.name) > 0:
                     fields.append(channel.s.name)
-        return {'fields':fields}
-    
-    def select_plot_channels(self,chan_names):
-        
+        return {'fields': fields}
+
+    def select_plot_channels(self, chan_names):
```

Comment:
Consider making `chan_names=None` so it is not necessary to call as `select_plot_channels(None)`, just `select_plot_channels()`